### PR TITLE
Add on redirect callback

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Adds a callback to component for merchants to get notified when we redirect to a browser, custom tab or external app.

## Checklist 
- [ ] Code is unit tested
- [x] Changes are tested manually
- [x] Add relevant labels to PR  

COAND-732

### CODE REVIEW

File: components-core/src/main/java/com/adyen/checkout/components/core/RedirectableActionComponent.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Created by oscars on 1/8/2023.
+ */
+
+package com.adyen.checkout.components.core
+
+interface RedirectableActionComponent {
+
+    /**
+     * Set a callback that will be called when an redirect is opened.
```
Comment (jreij) [inline components-core/src/main/java/com/adyen/checkout/components/core/RedirectableActionComponent.kt:14]: Typo: `a` redirect
Also maybe we can use "when a redirect is made" or something?

File: drop-in/src/main/java/com/adyen/checkout/dropin/BaseDropInServiceContract.kt [hunk 2]
```diff
@@ -89,4 +89,9 @@ interface BaseDropInServiceContract {
      * [DropInConfiguration.Builder.setAdditionalDataForDropInService] or null if nothing was set.
      */
     fun getAdditionalData(): Bundle?
+
+    /**
+     * This method is called when a redirect is opened.
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/BaseDropInServiceContract.kt:94]: Let's align this doc with the one in the components

File: drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt [hunk 3]
```diff
@@ -565,6 +566,8 @@ internal fun getComponentFor(
         else -> {
             throw CheckoutException("Unable to find stored component for type - ${storedPaymentMethod.type}")
         }
+    }.apply {
+        this.setOnRedirectListener(onRedirect)
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt:570]: Probably not

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/DefaultRedirectHandler.kt [hunk 4]
```diff
@@ -156,21 +172,31 @@ class DefaultRedirectHandler : RedirectHandler {
...
     private fun launchBrowser(context: Context, uri: Uri): Boolean {
+        @Suppress("SwallowedException")
         return try {
             val browserActivityIntent = Intent()
                 .setAction(Intent.ACTION_VIEW)
                 .addCategory(Intent.CATEGORY_BROWSABLE)
                 .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                 .setData(uri)
             context.startActivity(browserActivityIntent)
+            onRedirectListener?.invoke()
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/DefaultRedirectHandler.kt:183]: I left it like this after removing the specific redirect method, but calling it in one place is better indeed 👍

**Review by jreij — APPROVED** [review]

**Pull request merged**